python/test/unit/language/test_core.py

```diff
@@ -4354,18 +4354,32 @@ def kernel(Out):
 # -----------------------
 # test layout conversions
 # -----------------------
-# TODO: backend should be tested separately
 
-layouts = [
-    BlockedLayout([1, 16], [8, THREADS_PER_WARP // 8], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([1, 8], [2, THREADS_PER_WARP // 2], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([1, 4], [4, THREADS_PER_WARP // 4], [2, 2], [1, 0], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([1, 1], [1, THREADS_PER_WARP], [2, 2], [1, 0], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([8, 1], [16, THREADS_PER_WARP // 16], [1, 4], [0, 1], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([4, 1], [8, THREADS_PER_WARP // 8], [2, 2], [0, 1], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([1, 1], [THREADS_PER_WARP, 1], [2, 2], [0, 1], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([4, 4], [1, THREADS_PER_WARP], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1])
-]
+if is_hip():
+    # limit layouts to fit scratch buffers into 64K LDS
+    # to guarantee that scratch buffer is not too large:
+    # elemsPerThread * threadsPerWarp * warps <= 64 for every spatial dimension
```

Comment:
Can we have a check that does this math and skip instead of duplicating the configuration? That will make it harder to maintain the test and make sure all the interesting cases are handled.

Reply:
Also I'm curious why is the layout affecting the shared memory size? The amount of shared memory mostly depends on the tensor size. Is it because we end up with different padding pushing the size over the limit? Smaller tensors would still work with some of those layouts then?
I think it is important to be able to test all those different layouts even if we may have to skip few cases with the largest 128x128 case.

Reply:
ok, will do.
If you are ok with runtime checks, I can even relax this requirement

Reply:
> If you are ok with runtime checks, I can even relax this requirement

you mean checking in the test and skip based on parameters? If so yes that sounds good.

Reply:
> you mean checking in the test and skip based on parameters?

yes!
Please take a look at new implementation.
I hope I did not overengineering it

Reply:
> Also I'm curious why is the layout affecting the shared memory size?

let's consider example of conversion from `BlockedLayout([1, 16], [8, THREADS_PER_WARP // 8], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1])` (layout 0 in test) to `BlockedLayout([8, 1], [16, THREADS_PER_WARP // 16], [1, 4], [0, 1], [1, 1], [1, 1], [0, 1])` (layout 4 in test) and some large tensor of shape [128x128]

src layout has:
- `per thread shape` == [1, 16]
- `threads per warp` == [8, 8]
- `warps per CTA` == [4, 1]

dst layout has:
- `per thread shape` == [8, 1]
- `threads per warp` == [16, 4]
- `warps per CTA` == [1, 4]

In example above this minimal "tile" size for src layout is [1*8*4, 16*8*1] == [32, 128]; for dst layout it is [8*16*1, 1*4*4] == [128, 16]

So minimal scratch buffer will have shape [128, 128] + some padding in the fastest dimension (current implementation always adds this padding).

Reply:
yeah I think that'll do the trick. Maybe adding a brief extra comment on why we are considering int type and mentioning that the helper only gives an estimate on the shape. Thanks for doing that